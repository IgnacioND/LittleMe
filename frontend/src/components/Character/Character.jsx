import React from 'react';

import {
  View, Text, ImageBackground,
} from 'react-native';
import PropTypes from 'prop-types';

const backgroundImage = require('../../assets/images/gradient-background-banding.png');

const styles = {
  main: {
    flex: 1,

  },
  header: {
    flexDirection: 'row',
    backgroundColor: '#2d1b42',
    justifyContent: 'space-around',
    width: '100%',
    height: 100,
    position: 'absolute',
    top: 0,
  },
  headerText: {
    marginTop: 70,
    color: 'white',
  },
  title: {
    fontSize: 20,
    color: 'white',
  },
  image: {
    width: 210,
    height: 60,
  },
  backgroundImage: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  button: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonImage: {
    zIndex: -1,
  },
  inputs: {
    height: 40,
    width: 160,
    backgroundColor: 'yellow',
    borderWidth: 2,
    borderColor: 'black',
    borderRadius: 10,
    padding: 10,
    margin: 5,
  },
  absoluteView: {
    flex: 1,
    position: 'absolute',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'transparent',
  },
};
export default function Character() {
  return (
    <View style={styles.main}>
      <ImageBackground style={styles.backgroundImage} resizeMode="cover" source={backgroundImage}>
        <View style={styles.header}>
          <Text style={styles.headerText}>16/09/2021</Text>
          <Text style={styles.headerText}>Home</Text>
          <Text style={styles.headerText}>NEXT: MEETING</Text>
        </View>
        <Text style={styles.title}> Character</Text>
      </ImageBackground>
    </View>
  );
}
Character.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};
