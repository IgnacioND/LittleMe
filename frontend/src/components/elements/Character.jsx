/* eslint-disable global-require */
import React from 'react';
import PropTypes from 'prop-types';
import {
  View, Image, Text,
} from 'react-native';
import { useSelector } from 'react-redux';
import * as Progress from 'react-native-progress';
import taskStyles from '../styles/taskStyles';

const characterMock = require('../../assets/images/characterMock.png');

const taskGroup = {
  health: require('../../assets/images/iconHealth.png'),
  mind: require('../../assets/images/iconMind.png'),
  money: require('../../assets/images/iconMoney.png'),
  hygiene: require('../../assets/images/iconHygiene.png'),
};
export default function Character({ navigation }) {
  const { little } = useSelector((store) => store.user);
  const characterHandler = () => {
    navigation.navigate('CharacterScreen');
  };
  const stats = {
    hunger: (little?.stats?.hunger > 0) ? little.stats.hunger / 10 : 0,
    money: (little?.stats?.money > 0) ? little.stats.money / 100 : 0,
    hygiene: (little?.stats?.hygiene > 0) ? little.stats.hygiene / 10 : 0,
    mind: (little?.stats?.mind > 0) ? little.stats.mind / 10 : 0,
  };
  // useEffect(() => {}, [little]);
  return (
    <View>
      <View style={{ flexDirection: 'row', justifyContent: 'space-around', margin: 5 }}>
        <Image
          source={taskGroup.health}
          style={[taskStyles.healthIcon, { width: 25, height: 25 }]}
        />
        <Progress.Bar progress={stats.hunger} width={100} height={25} color="black" />
        <Progress.Bar progress={stats.money} width={100} height={25} color="black" />
        <Image
          source={taskGroup.money}
          style={[taskStyles.healthIcon, { width: 25, height: 25 }]}
        />
      </View>
      <View style={{ flexDirection: 'row', justifyContent: 'space-around', margin: 5 }}>
        <Image
          source={taskGroup.mind}
          style={[taskStyles.healthIcon, { width: 25, height: 25 }]}
        />
        <Progress.Bar progress={stats.mind} width={100} height={25} color="black" />
        <Progress.Bar progress={stats.hygiene} width={100} height={25} color="black" />
        <Image
          source={taskGroup.hygiene}
          style={[taskStyles.healthIcon, { width: 25, height: 25 }]}
        />
      </View>

      <Image
        style={{ height: 250, width: 340 }}
        // style={characterStyles.characterBack}
        source={characterMock}
        onPress={characterHandler}
      />
    </View>

  );
}

Character.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};
