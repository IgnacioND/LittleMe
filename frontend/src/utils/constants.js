export const MONEY_INCREMENT = 10;
export const STANDARD_INCREMENT = 1;
export const MONEYPLUS_INCREMENT = MONEY_INCREMENT * 2;
export const NULL_INCREMENT = 0;
