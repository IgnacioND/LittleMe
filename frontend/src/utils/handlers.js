// import { useDispatch } from 'react-redux';

export const goToTasksScreen = (navigation) => () => navigation.navigate('TasksScreen');
export const goToLoginScreen = (navigation) => () => navigation.navigate('LoginScreen');
export const goToSignupScreen = (navigation) => () => navigation.navigate('SignupScreen');
export const goToDashboardScreen = (navigation) => () => navigation.navigate('DashboardScreen');
export const goToProfileScreen = (navigation) => () => navigation.navigate('ProfileScreen');
export const goToCalendarScreen = (navigation) => () => navigation.navigate('CalendarScreen');
export const goToCharacterScreen = (navigation) => () => navigation.navigate('CharacterScreen');
export const goToCreateTaskScreen = (navigation) => () => navigation.navigate('CreateTaskScreen');
export const goToHomeScreen = (navigation) => () => navigation.navigate('HomeScreen');
export const goToScreen = (navigation) => () => navigation.navigate('Screen');
