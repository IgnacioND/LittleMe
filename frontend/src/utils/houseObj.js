const houseObj = [
  {
    name: 1,
    leftWall: 1,
    rightWall: 1,
    centerWall: 1,
    floor: 1,
    ceiling: 1,
    leftFurniture: 1,
    centerFurniture: 1,
    rightFurniture: 1,
    tidynessLevel: 1,
  },
];

module.exports = houseObj;
