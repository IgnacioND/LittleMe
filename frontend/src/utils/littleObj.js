const littleObj = {
  name: 'Pepe',
  age: 1,
  evolution: 1,
  skin: 1,
  appeareance: {
    eyes: 1,
    mouth: 1,
    hair: 1,
    extra1: 1,
    extra2: 1,
    height: 1,
    width: 1,
  },
  stats: {
    money: 3,
    hunger: 5,
    hygiene: 5,
    mind: 3,
  },
  clothes: {
    shirt: 1,
    pants: 1,
    shoes: 1,
  },
  tendencies: {
    studies: 1,
    healthiness: 1,
    leisure: 1,
  },
};
module.exports = littleObj;
