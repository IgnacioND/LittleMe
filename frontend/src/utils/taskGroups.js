const taskGroups = {
  work: ['reunion', 'meeting', 'colleage', 'worker', 'company',
    'coworker', 'co-worker', 'working', 'work', 'project'],
  studies: ['read', 'homework', 'book', 'learn', 'study', 'degree', 'master'],
  house: [
    'kitchen', 'living-room', 'room', 'bathroom',
    'buy', 'clean', 'cleaning', 'dormitory', 'hall',
    'cook', 'groceries', 'bleach', 'ironing', 'detergent',
    'laundry', 'dust', 'washing', 'floor', 'mop',
    'lawn', 'rubbish', 'tidy', 'vacuum', 'dishes', 'broom', 'chores'],
  leisure: ['friend', 'movie', 'movies', 'cinema', 'hangout',
    'hobby', 'holiday', 'relax', 'relaxation', 'rest', 'casual',
    'enjoy', 'travel', 'vacation', 'play', 'theater', 'hang out'],
  health: ['gym', 'dance', 'run', 'doctor', 'hair', 'football', 'play', 'dojo', 'fight',
    'lunch', 'medical', 'therapy', 'psycology', 'shrunk', 'eat', 'dinner', 'sleep',
    'breackfast', 'dinner', 'lunch'],

};
module.exports = taskGroups;
