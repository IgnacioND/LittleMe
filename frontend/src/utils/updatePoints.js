import {
  MONEY_INCREMENT,
  STANDARD_INCREMENT,
  MONEYPLUS_INCREMENT,
  NULL_INCREMENT,
} from './constants';

export default function updatePoints(task, completeUser) {
  const isWorkTask = task === 'work';
  const isHealthTask = task === 'health';
  const isLeisureTask = task === 'leisure';
  const isMindTask = ['study', 'house', 'leisure'].includes(task);
  const isHealthyTask = ['study', 'house', 'health'].includes(task);
  const isWorkStudyTask = ['study', 'work'].includes(task);
  const { stats, tendencies } = completeUser?.little || {};

  return {
    little: {
      stats: {
        money: stats?.money + (isWorkTask ? MONEYPLUS_INCREMENT : MONEY_INCREMENT),
        hunger: stats?.hunger + STANDARD_INCREMENT,
        hygiene: stats?.hygiene + (isHealthTask ? STANDARD_INCREMENT : NULL_INCREMENT),
        mind: stats?.mind + (isMindTask ? STANDARD_INCREMENT : NULL_INCREMENT),

      },
      tendencies: {
        studies: tendencies?.studies + (isWorkStudyTask ? STANDARD_INCREMENT : NULL_INCREMENT),
        healthiness: tendencies?.healthiness + isHealthyTask ? STANDARD_INCREMENT : NULL_INCREMENT,
        leisure: tendencies?.leisure + (isLeisureTask ? STANDARD_INCREMENT : NULL_INCREMENT),
      },
    },
  };
}
