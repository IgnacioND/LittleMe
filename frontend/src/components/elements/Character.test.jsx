import React from 'react';
import { render } from '../../utils/test.utils';
import Character from './Character';

const navigation = {
  navigate: jest.fn(),
};

describe('Given a Character Component', () => {
  describe('When it is rendered', () => {
    let screen;
    beforeEach(() => {
      screen = render(<Character navigation={navigation} />);
    });
    test('Then screen should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
});
