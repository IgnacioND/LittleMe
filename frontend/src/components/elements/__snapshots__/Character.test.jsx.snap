// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Character Component When it is rendered Then screen should match the snapshot 1`] = `
<View>
  <Text>
    money: undefined
  </Text>
  <Text>
    hunger: undefined
  </Text>
  <Text>
    hygiene: undefined
  </Text>
  <Text>
    mind: undefined
  </Text>
  <Text>
    health: undefined
  </Text>
  <Text>
    studies: undefined
  </Text>
  <Text>
    leisure: undefined
  </Text>
  <Image
    onPress={[Function]}
    source={1}
    style={
      Object {
        "height": 250,
        "width": 340,
      }
    }
  />
</View>
`;
